.vscode/settings.default.jsonc

```diff
@@ -0,0 +1,62 @@
+{
+  // Please keep this file in sync with settings in home-assistant/.devcontainer/devcontainer.json
+  // common
+  "editor.formatOnPaste": false,
+  "editor.formatOnSave": true,
+  "editor.formatOnSaveTimeout": 3000,
+  "editor.formatOnType": true,
+  "editor.insertSpaces": true,
+  "files.insertFinalNewline": true,
+  "files.trimTrailingWhitespace": true,
```

Comment:
This does not apply to all file types?

Reply:
That setting already existed: https://github.com/home-assistant/core/blob/dev/.devcontainer/devcontainer.json#L27

Reply:
I think the counter to that is name a language in HA that you would not want to trim trailing whitespace:

* Python: yes
* JSON: yes
* Dockerfile: yes
* Yaml: yes

Trailing whitespace creates inconsistent diffs if someone is not getting rid of it.

Reply:
Markdown: No
Fixtures: No

Reply:
Fixtures are just JSON no? Also, why would you not want to trim either?

Reply:
A fixture can be invalid on purpose for tests too...

Reply:
That is a great point. But wouldn't you want the 99% case to be the one everyone gets? You can just do Command Palette -> Save without Formatting if you need to override an auto-formatter's behavior.